dashboard/app/models/school_info.rb

```diff
@@ -73,6 +74,51 @@ def school_state=(input)
     end
   end
 
+  # Only sync from school on create to avoid unintended updates to old data
+  before_validation :sync_from_schools, on: :create
+
+  def sync_from_schools
+    # If a SchoolInfo is linked to a School then the SchoolInfo pulls its data from the School
+    # It seems like there is some code that is passing in mismatched data at times.
+    # As of Nov. 2, 2017 there were 7 rows in school_infos with non-null school_id and conflicting
+    # data between schools and school_infos. Until we better understand what is causing that we don't want
+    # to flat out fail if we get conflicting inputs. Instead we overwrite the passed in fields with what
+    # is on the School and report the mismatch to HoneyBadger.
+    unless school.nil? && school_id.nil?
+      original = {}
+      original[:country] = country
+      original[:school_type] = school_type
+      original[:state] = state
+      original[:zip] = zip
+      original[:school_district_id] = school_district_id
+      original[:school_district_other] = school_district_other
+      original[:school_district_name] = school_district_name
+      original[:school_other] = school_other
+      original[:school_name] = school_name
+      original[:full_address] = full_address
+      original[:validation_type] = validation_type
+
+      school = School.find(school_id) if school.nil?
+      self.country = 'US' # Everything in SCHOOLS is a US school
+      self.school_type = school.school_type
+      self.state = school.state
+      self.zip = nil
+      self.school_district = school.school_district
+      self.school_district_other = nil
+      self.school_district_name = nil
+      self.school_other = nil
+      self.school_name = nil
+      self.full_address = nil
+      self.validation_type = VALIDATION_FULL
+
+      # Report if we are overriding a non-nil value that was originally passed in
+      something_overwritten = original.map {|key, value| value && value != self[key]}.reduce {|acc, b| acc || b}
+      if something_overwritten
```

Comment:
If you use the above approach, and store the keys in an array, then you should just be able to compare hashes here:

```ruby
# from above
relevant_attributes = %w(
  country
  school_type
) # etc

# here:
if original != attributes.slice(*relevant_attributes)
  # log to honeybadger
```

Reply:
I don't think that works since I don't want to compare nils from original with anything in the final set of attributes.